File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        Set<Integer> set =new HashSet<>();
        for(int it: nums){
            set.add(it);
        }

        ListNode dummy =new ListNode();
        ListNode curr =dummy;
        ListNode trav =head;
           
        while(trav!=null){
            if(!set.contains(trav.val)){
               curr.next =trav;
               curr =curr.next;
            }
            trav =trav.next;
        }
       curr.next =null;
         
        return dummy.next;
    }
}

File: 3501-delete-nodes-from-linked-list-present-in-array/3501-delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        if head ==None:
            return head
        dummy =ListNode(0)
        dummy.next =head

        prev =dummy
        curr =head

        nums_set =set(nums)

        while curr is not None:
            if curr.val in nums_set:
                prev.next =curr.next
            else:
                prev =curr
            curr= curr.next
        return dummy.next
